src/components/TupleSelect/index.tsx

```diff
@@ -20,37 +18,43 @@ type option = {
 interface TupleSelectProps {
   options: option[];
   onSubmit?: ({ category, value }: { category: string; value: string }) => void;
+  validator?: (value: string) => boolean;
+  validatorErrorMessage?: string;
 }
-export const TupleSelect: React.VFC<TupleSelectProps> = ({
+const TupleSelect: React.VFC<TupleSelectProps> = ({
   options,
   onSubmit = () => {},
+  validator = () => true,
+  validatorErrorMessage = "Invalid Input",
 }) => {
   const [input, setInput] = useState("");
   const [selected, setSelected] = useState(options[0].value);
-  const updateQueryParams = useUpdateURLQueryParams();
-  const { search } = useLocation();
-  const queryParams = parseQueryString(search);
+  const isValid = validator(input);
 
   const handleOnSubmit = () => {
-    const selectedParams = queryParams[selected] as string[];
-    const updatedParams = upsertQueryParam(selectedParams, input);
-    onSubmit({ category: selected, value: input });
-    updateQueryParams({ [selected]: updatedParams });
-    setInput("");
+    if (isValid) {
+      onSubmit({ category: selected, value: input });
+      setInput("");
+    }
+  };
+
+  const handleOnChange = (value: string) => {
+    setInput(value);
   };
   const selectedOption = options.find((o) => o.value === selected);
+
   return (
     <Container>
       <Label htmlFor="filter-input">
         Add New {selectedOption.displayName} Filter
       </Label>
-      <Input.Group compact>
-        <Select
-          style={{ width: "30%" }}
+      <InputGroup>
+        <GroupedSelect
```

Comment:
It doesn't like when I click on the placeholder.

https://user-images.githubusercontent.com/47064971/165418048-1607b5f9-e87f-4dc9-95af-647b5a05b385.mov

You can get rid of the placeholder by adding the props `allowDeselect={false}` though, which will make this problem go away :)

Reply:
Nice catch!